molgenis-js/src/main/java/org/molgenis/js/nashorn/NashornScriptEngine.java

```diff
@@ -132,23 +92,15 @@ private Object convertNashornValue(Object nashornValue)
 					JsDate jsDate = ((Invocable) scriptEngine).getInterface(scriptObjectMirror, JsDate.class);
 					return jsDate.getTime();
 				}
-				else if (((ScriptObjectMirror) nashornValue).containsKey(idValueKey))
-				{
-					// entity object returned from script
-					return ((ScriptObjectMirror) nashornValue).get(idValueKey);
-				}
-				else
-				{
-					throw new RuntimeException("Unable to convert [ScriptObjectMirror]");
-				}
+				else return ((ScriptObjectMirror) nashornValue).getOrDefault(KEY_ID_VALUE, null);
```

Comment:
Why change throwing an exception to return ```null```?

Reply:
It happens at initialisation time when the libraries are loaded. They return the ScriptObjectMirror for a javascript function. So the choice here was to return the ScriptObjectMirror or to toss the mirror away.